src/Psl/DateTime/Internal/zone_override.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\DateTime\Internal;
+
+use Closure;
+use Psl\DateTime\Timezone;
+
+use function date_default_timezone_get;
+use function date_default_timezone_set;
+
+/**
+ * @template T
+ *
+ * @param (Closure(): T) $callback
+ *
+ * @return T
+ *
+ * @internal
+ *
+ * @pure - This technically a lie.
+ */
+function zone_override(?Timezone $tz, Closure $callback): mixed
```

Comment:
I don't really like this function and have the feeling it is used as some kind of workaround in a lot of scenarios.

Reply:
removed. we now use IntlTimezone/IntlCalendar/IntlDateFormatter internally, none of which need global state, so our code is not truly immutable :D